Geometry/GEMGeometryBuilder/src/ME0GeometryBuilder.cc

```diff
@@ -0,0 +1,518 @@
+
+// ==========================================
+// ===  Test to understand the structure  ===
+// ==========================================
+#ifdef EDM_ML_DEBUG
+  bool testChambers = fv.firstChild();
+  LogTrace("ME0GeometryBuilder") << "doChamber = fv.firstChild() = " << testChambers;
+
+  while (testChambers) {
+    // to etapartitions
+    LogTrace("ME0GeometryBuilder") << "to layer " << fv.firstChild();
+    LogTrace("ME0GeometryBuilder") << "to etapt " << fv.firstChild();
+    MuonDDDNumbering mdddnum(muonConstants);
+    ME0NumberingScheme me0Num(muonConstants);
+    int rawId = me0Num.baseNumberToUnitNumber(mdddnum.geoHistoryToBaseNumber(fv.geoHistory()));
+    ME0DetId detId = ME0DetId(rawId);
+    ME0DetId detIdCh = detId.chamberId();
+    // back to chambers
+    LogTrace("ME0GeometryBuilder") << "back to layer " << fv.parent();
+    LogTrace("ME0GeometryBuilder") << "back to chamb " << fv.parent();
+
+    LogTrace("ME0GeometryBuilder") << "In DoChambers Loop :: ME0DetId " << detId << " = " << detId.rawId()
+                                   << " (which belongs to ME0Chamber " << detIdCh << " = " << detIdCh.rawId() << ")";
+    LogTrace("ME0GeometryBuilder") << "Second level logical part: " << fv.logicalPart().name().name();
+    DDBooleanSolid solid2 = (DDBooleanSolid)(fv.logicalPart().solid());
+    std::vector<double> dpar2 = solid2.parameters();
+    std::stringstream parameters2;
+    for (unsigned int i = 0; i < dpar2.size(); ++i) {
+      parameters2 << " dpar[" << i << "]=" << dpar2[i] / 10 << "cm ";
+    }
+    LogTrace("ME0GeometryBuilder") << "Second level parameters: vector with size = " << dpar2.size() << " and elements "
+                                   << parameters2.str();
+
+    bool doLayers = fv.firstChild();
+
+    LogTrace("ME0GeometryBuilder") << "doLayer = fv.firstChild() = " << doLayers;
+    while (doLayers) {
+      // to etapartitions
+      LogTrace("ME0GeometryBuilder") << "to etapt " << fv.firstChild();
+      MuonDDDNumbering mdddnum(muonConstants);
+      ME0NumberingScheme me0Num(muonConstants);
+      int rawId = me0Num.baseNumberToUnitNumber(mdddnum.geoHistoryToBaseNumber(fv.geoHistory()));
+      ME0DetId detId = ME0DetId(rawId);
+      ME0DetId detIdLa = detId.layerId();
+      // back to layers
+      LogTrace("ME0GeometryBuilder") << "back to layer " << fv.parent();
+      LogTrace("ME0GeometryBuilder") << "In DoLayers Loop :: ME0DetId " << detId << " = " << detId.rawId()
+                                     << " (which belongs to ME0Layer " << detIdLa << " = " << detIdLa.rawId() << ")";
+      LogTrace("ME0GeometryBuilder") << "Third level logical part: " << fv.logicalPart().name().name();
+      DDBooleanSolid solid3 = (DDBooleanSolid)(fv.logicalPart().solid());
+      std::vector<double> dpar3 = solid3.parameters();
+      std::stringstream parameters3;
+      for (unsigned int i = 0; i < dpar3.size(); ++i) {
+        parameters3 << " dpar[" << i << "]=" << dpar3[i] / 10 << "cm ";
```

Comment:
```suggestion
        parameters3 << " dpar[" << i << "]=" << convertMmToCm(dpar3[i]) << "cm ";
```

Reply:
Same change for lines 135, 231-234, 240, 258-261, 267 301-305, 310, and 337. We are trying to eliminate CLHEP units in CMSSW.

Reply:
@cvuosalo : done